Update Airflow version references to `3.X` in `README_RELEASE_PROVIDER.md`

jroachgolf84 opened the pull request:
Updating references to Airflow `2.X` in `README_RELEASE_PROVIDER.md` to now point to Airflow `3.X`, where appropriate.

jroachgolf84 committed:
Update Airflow version references to `3.x` in `README_RELEASE_PROVIDERS.md`

amoghrajesh commented on dev/README_RELEASE_PROVIDERS.md:
```diff
@@ -1018,9 +1018,8 @@ Provider distributions. This is especially helpful when you want to test integra
-
-RUN pip install  --user apache-airflow-providers-google==2.2.2.rc1
+FROM apache/airflow:3.1.3
+RUN pip install  --user apache-airflow-providers-google==18.1.0.rc1
```
I think this version can be confusing to readers, should we mention that it van be arbitrary?

jroachgolf84 replied:
I think that makes sense. I'll update this now.

jroachgolf84 committed:
Clarified instructions for building a custom Docker image with Google provider

Clarified instructions for building a custom Docker image with Google provider.

jroachgolf84 committed:
Merge branch 'main' into patch-7

jedcunningham commented on dev/README_RELEASE_PROVIDERS.md:
```diff
@@ -1015,12 +1015,13 @@ Provider distributions is used.
 If you prefer to build your own image, you can also use the official image and PyPI packages to test
 Provider distributions. This is especially helpful when you want to test integrations, but you need to install
-additional tools. Below is an example Dockerfile, which installs providers for Google/
+additional tools. Below is an example Dockerfile, which installs providers for Google. Please note, these 
```
```suggestion
additional tools. Below is an example Dockerfile, which installs providers for Google. Please note, these
```

jedcunningham replied:
@jroachgolf84 make sure you have prek hooks running locally 😅

jedcunningham committed:
Update dev/README_RELEASE_PROVIDERS.md
